tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/IR/hlo_ops.cc

```diff
@@ -1351,6 +1351,27 @@ static LogicalResult Verify(BroadcastOp op) {
   return success();
 }
 
+OpFoldResult BroadcastOp::fold(ArrayRef<Attribute> attrs) {
+  auto type = getType().cast<RankedTensorType>();  
+  if (type == getOperand().getType()) {
+    auto sizesSize = broadcast_sizes().getType().getNumElements();
+    if (sizesSize != 0) {
+      return {};
+    }
+    return getOperand();
+  }
+
+  // Constant fold when an operand is a splat tensor attribute.
+  if (!attrs[0] || !type.hasStaticShape()) return {};
+  auto splatOperandAttr = attrs[0].dyn_cast<SplatElementsAttr>();
+  if (!splatOperandAttr) return {};
```

Comment:
Can you add test for these three conditions?

Reply:
done

Reply:
I think you're missing a case for the static shape test?

Reply:
I can't create a constant value with dynamic shape, so I add dynamic shape identity test case instead.

Reply:
Yes that's what I was looking for, thanks :)